### PR TITLE
Pending discussion: configurable validity period handling

### PR DESCRIPTION
**Pull request by ties**
A WIP for configurable validity period handling and rejection of objects from the future.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -46,6 +46,16 @@ provisioning protocol.
 
 ## Changelog
 
+### 2019-02 version 1.8-SNAPSHOT
+
+ - Fix: Warn for CRLs with nextUpdate in grace period, reject CRLs with
+   nextUpdate outside grace period.
+ - Fix: Warn for manifests with nextUpdate in grace period, reject manifests
```
Comment (wibisono) [inline README.md:53]: From the code you also add a Warning if this update time is in the future.

Comment (wibisono) [reply README.md:53]: nvm misread.

File: src/test/java/net/ripe/rpki/commons/crypto/cms/manifest/ManifestCmsTest.java [hunk 2]
```diff
@@ -202,7 +203,7 @@ public void shouldWarnWhenManifestIsStale() {
         CertificateRepositoryObjectValidationContext context = new CertificateRepositoryObjectValidationContext(ROOT_CERTIFICATE_LOCATION, rootCertificate, resources, Lists.newArrayList(rootCertificate.getSubject().getName()));
 
         ValidationOptions options = new ValidationOptions();
-        options.setMaxStaleDays(Integer.MAX_VALUE);
+        options.setManifestMaxStalePeriod(Duration.standardDays(Integer.MAX_VALUE));
```
[the remark this answers is not in the document]
Comment (ties) [reply src/test/java/net/ripe/rpki/commons/crypto/cms/manifest/ManifestCmsTest.java]: A less radical value would be a safer pick (I took 100 years).

**Comment by ties** [comment]
rebased from master. Should be good now.

File: src/main/java/net/ripe/rpki/commons/validation/ValidationOptions.java [hunk 3]
```diff
@@ -30,21 +30,37 @@
...
 /**
  * User controlled options to use when validating objects.
  */
 public class ValidationOptions {
-
-    private int maxStaleDays = 0;
+    private Duration crlMaxStalePeriod = Duration.ZERO;
+    /**
+     * Grace period for the NEXT_UPDATE_TIME of Manifest. When a manifest is in the grace period, the manifest causes
+     * a warning on validation instead of an failure.
```
Comment (lolepezy) [inline src/main/java/net/ripe/rpki/commons/validation/ValidationOptions.java:42]: "a failure" I guess

**Pull request merged**